elastica/contact_forces.py

```diff
@@ -725,3 +861,120 @@ def apply_contact(
+        friction_coefficient=0.0,
+    ):
+        """
+        Parameters
+        ----------
+        k : float
+            Contact spring constant.
+        nu : float
+            Contact damping constant.
+        velocity_damping_coefficient : float
+            Velocity damping coefficient between rigid-body and rod contact is used to apply friction force in the
+            slip direction.
+        friction_coefficient : float
+            For Coulombic friction coefficient for rigid-body and rod contact.
+        """
+        super(RodSphereContact, self).__init__()
+        self.k = k
+        self.nu = nu
+        self.velocity_damping_coefficient = velocity_damping_coefficient
+        self.friction_coefficient = friction_coefficient
+
+    def _check_systems_validity(
+        self,
+        system_one: SystemType,
+        system_two: AllowedContactType,
+    ) -> None:
+        """
+        This checks the contact order and type of a SystemType object and an AllowedContactType object.
+        For the RodSphereContact class first_system should be a rod and second_system should be a sphere.
+        Parameters
+        ----------
+        system_one
+            SystemType
+        system_two
+            AllowedContactType
+        """
+        if not issubclass(system_one.__class__, RodBase) or not issubclass(
+            system_two.__class__, Sphere
+        ):
+            raise TypeError(
+                "Systems provided to the contact class have incorrect order/type. \n"
+                " First system is {0} and second system is {1}. \n"
+                " First system should be a rod, second should be a sphere".format(
+                    system_one.__class__, system_two.__class__
+                )
+            )
+
+    def apply_contact(self, system_one: RodType, system_two: AllowedContactType):
+        # First, check for a global AABB bounding box, and see whether that
+        # intersects
+        if _prune_using_aabbs_rod_sphere(
+            system_one.position_collection,
+            system_one.radius,
+            system_one.lengths,
+            system_two.position_collection,
+            system_two.director_collection,
+            system_two.radius[0],
+        ):
+            return
+
+        x_sph = (
+            system_two.position_collection[..., 0]
+            - system_two.radius * system_two.director_collection[2, :, 0]
+        )
+
+        rod_element_position = 0.5 * (
+            system_one.position_collection[..., 1:]
+            + system_one.position_collection[..., :-1]
+        )
+        _calculate_contact_forces_rod_sphere(
+            rod_element_position,
+            system_one.lengths * system_one.tangents,
+            system_two.position_collection[..., 0],
+            x_sph,
+            system_two.radius * system_two.director_collection[2, :, 0],
+            system_one.radius + system_two.radius,
+            system_one.lengths + system_two.lengths,
```

Comment:
Length doesn't make much sense of sphere, and it is set as length = 2*radius. Maybe here we replace with following 
```suggestion
            system_one.lengths + 2 * system_two.radius,
```

Reply:
I agree, maybe we should remove it later too cause it only makes sense for cylinders.